### ISSUE TITLE
Calculate C-Index with Pytorch

### ISSUE THREAD

**Issue by Minxiangliu** [MGensheimer/nnet-survival#5]
Hi @MGensheimer,
If I want to calculate the c-index value for the test data in this PyTorch [example](https://github.com/MGensheimer/nnet-survival/blob/master/nnet_survival_pytorch_example.ipynb), how should I do it?

If I want to determine the c-index at the maximum time, do I just need to do the following?
```py
surv_pred=np.cumprod(y_pred, axis=1)[:,-1]
concordance_index(times, surv_pred, events)
```

Additionally, I would like to ask, if I want to save the best model during the training, should I prioritize the validation loss or the c-index?

**Comment by MGensheimer** [comment]
For C-index, you can use similar code to https://github.com/MGensheimer/nnet-survival/blob/master/nnet_survival_examples.py

For the PyTorch example, each individual's time to censoring or failure is kept in the variable t, and whether they had an event (also known as a failure) is kept in the variable f.

So, first, calculate the model-predicted probability of each individual surviving more than 1 year:

```
oneyr_surv=np.cumprod(y_pred[:,0:np.nonzero(breaks>365)[0][0]], axis=1)[:,-1]
```

Then, calculate the C-index using the t and f variables to indicate each individual's actual survival outcome:

```
print(concordance_index(t,oneyr_surv,f))
```

For saving the best model, that would depend on the use case but generally I would suggest using validation loss since that incorporates both discrimination and calibration whereas the C-index only captures discrimination and you can have a very poorly calibrated model that over-  or under-predicts survival that has a good C-index.
